lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1969,6 +1969,30 @@ static SILValue getTensorProtocolHandleMember(SILValue v, SILLocation loc,
   return v;
 }
 
+/// Given a type, recursively collect all innermost element types if it's an
+/// aggregate of TensorHandle's or dtypes.
+static void collectInnermostTensorFlowDTypes(
+    CanType type, SmallVectorImpl<SymbolicValue> &result) {
+  if (isTensorFlowDType(type)) {
+    result.push_back(SymbolicValue::getMetatype(type));
+  } else if (tf::isTensorHandle(type)) {
+    auto eltType = getTensorHandleElementType(type)->getCanonicalType();
```

Comment:
Add an assertion below `assert(tf::isTensorFlowDType(eltType))` to make sure this is not a generic type parameter type or anything but a dtype.

Reply:
Done